cradlepoint/util/addEntry.js

```diff
@@ -32,3 +31,40 @@ export async function addTest(data) {
         throw err
     }
 }
+
+export async function addTestCase(data) {
+    try {
+        const client = await connectToDb();
+        const valid = await testCaseSchema.isValid(data)
+        if (valid && ObjectId.isValid(data.testPlanId) ) {
+            const testCase = testCaseSchema.cast(data);
+            for (const i in testCase.tests) {
```

Comment:
Ditch the for loop, replace with my code from EditTestCase in my PR. Make sure to also check that device Ids are correct in the BOM.

Reply:
again, I don't that that is actually better. This will provide a more descriptive error than your map function while still validating the data